File: Battleships_in_a_Board/Battleships_in_a_Board.go
package main

import "fmt"

func main() {
	board := [][]string{{"X", ".", ".", "X"}, {".", ".", ".", "X"}, {".", ".", ".", "X"}}
	fmt.Println(countBattleships(board))

}

func countBattleships(board [][]string) int {
	res := 0
	for i := 0; i < len(board); i++ {
		for j := 0; j < len(board[i]); j++ {
			if board[i][j] == "." {
				continue
			}

			if i > 0 && board[i-1][j] == "X" {
				continue
			}
			if j > 0 && board[i][j-1] == "X" {
				continue
			}
			res++
		}
	}

	return res
}

File: Battleships_in_a_Board/Battleships_in_a_Board.rs
fn main(){
    let board:Vec<Vec<String>> = vec![vec!["X".to_string(),".".to_string(),".".to_string(),"X".to_string()],vec![".".to_string(),".".to_string(),".".to_string(),"X".to_string()],vec![".".to_string(),".".to_string(),".".to_string(),"X".to_string()]];
    println!("{:?} ",countBattleships(board));
}

fn countBattleships(board:Vec<Vec<String>>) ->i32{
    let mut res = 0;
    for x in 0..board.len(){
        for y in 0..board[x].len(){
            if board[x][y] == ".".to_string(){
                continue
            } 
            if x > 0 && board[x-1][y] == "X".to_string(){
                continue
            }
            if y > 0 && board[x][y-1] == "X".to_string(){
                continue
            }
            res +=1; 
        }
    } 
    println!("{} ",res);
    return 0;
}